plugins/removeDeprecatedAttrs.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const { attrsGroupsDeprecated, elems } = require('./_collections');
+
+exports.name = 'removeDeprecatedAttrs';
+exports.description = 'removes deprecated attributes';
+
+/**
+ * Remove deprecated attributes.
+ *
+ * @type {import('./plugins-types').Plugin<'removeDeprecatedAttrs'>}
+ */
+exports.fn = (root, params) => {
+  const { removeUnsafe = false } = params;
+
+  return {
+    element: {
+      enter: (node) => {
+        const elemConfig = elems[node.name];
+        if (!elemConfig) {
+          return;
+        }
+
+        elemConfig.attrsGroups.forEach((attrsGroup) => {
+          const deprecated = attrsGroupsDeprecated[attrsGroup];
+          if (!deprecated) {
+            return;
+          }
+
+          if (deprecated.safe) {
+            deprecated.safe.forEach((name) => {
+              delete node.attributes[name];
+            });
+          }
+
+          if (removeUnsafe && deprecated.unsafe) {
+            deprecated.unsafe.forEach((name) => {
+              delete node.attributes[name];
+            });
+          }
+        });
+
+        const deprecated = elemConfig.deprecated;
+        if (!deprecated) {
+          return;
+        }
+
+        if (deprecated.safe) {
+          deprecated.safe.forEach((name) => {
+            delete node.attributes[name];
+          });
+        }
+
+        if (removeUnsafe && deprecated.unsafe) {
+          deprecated.unsafe.forEach((name) => {
+            delete node.attributes[name];
+          });
+        }
```

Comment:
Might be better to have a function like:

```js
function processAttributes(node, deprecatedAttrs, params) {
  if (!deprecatedAttrs) {
    return;
  }

  if (deprecatedAttrs.safe) {
    deprecatedAttrs.safe.forEach((name) => {
      delete node.attributes[name];
    });
  }

  if (params.removeUnsafe && deprecatedAttrs.unsafe) {
    deprecatedAttrs.unsafe.forEach((name) => {
      delete node.attributes[name];
    });
  }
}
```


```suggestion
        elemConfig.attrsGroups.forEach((attrsGroup) => {
          processAttributes(node, attrsGroupsDeprecated[attrsGroup], params);
        });
        
        processAttributes(node, elemConfig.deprecated, params);
```